docs/en/observability/logs-stream.asciidoc

```diff
@@ -133,20 +133,299 @@ image::images/stream-logs-example.png[example search query on the logs stream pa
+        "field": "message",
+        "pattern": "%{@timestamp} %{message}"
+      }
+    },
+    {
+      "date": {
+        "field": "@timestamp",
+        "formats": ["ISO8601"]
+      }
+    }
+  ]
+}
+----
+
+. Set these values for the dissect processor:
+- `field` – The field you're extracting data from, `message` in this case.
+- `pattern`– The pattern of the elements in your log data:
++
+[source,JSON]
+----
+%{timestamp} %{message}
+----
++
+This extracts the timestamp, `2023-08-08T13:45:12.123Z`, to the `timestamp` field, and the rest of the message, `WARN 192.168.1.101 Disk usage exceeds 90%.`, stays in the `message` field.
++
+. Set these values for the date processor:
+- `_ingest/pipeline/logs-example-default` – The name of the pipeline,`logs-example-default`, needs to match the name of your data stream. You'll set up your data stream in the next section. See {fleet-ref}/data-streams.html#data-streams-naming-scheme[data stream naming scheme] for more information.
+- `field` – The field you're extracting the date from, `@timestamp` in this case.
+- `formats` – The format of the current timestamp, `ISO8601` in this case. The date processor accepts {ref}/mapping-date-format.html[Java time patterns] or the following formats: `ISO8601`, `UNIX`, `UNIX_MS`, `TAI64N`.
+
+You can set additional fields with the date processor to update timezones, set a different output field, and change the output format of your timestamp. For more on this, see the {ref}/date-processor.html[date processor] documentation.
+
+[discrete]
+[[logs-stream-simulate-api]]
+==== Test your pipeline with the simulate pipeline API
+
+Make sure your pipeline is working as expected with the {ref}/simulate-pipeline-api.html#ingest-verbose-param[simulate pipeline API]. Run this command to test your pipeline:
+
+[source,console]
+----
+POST _ingest/pipeline/logs-example-default/_simulate
+{
+  "docs": [
+    {
+      "_source": {
+        "message": "2023-08-08T13:45:12.123Z WARN 192.168.1.101 Disk usage exceeds 90%."
+      }
+    }
+  ]
+}
+----
+
+The results should show the `@timestamp` field extracted from the `message` field:
+
+[source,console]
+----
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "_index",
+        "_id": "_id",
+        "_version": "-3",
+        "_source": {
+          "message": "WARN 192.168.1.101 Disk usage exceeds 90%.",
+          "@timestamp": "2023-08-08T13:45:12.123Z"
+        },
+        "_ingest": {
+          "timestamp": "2023-08-10T20:37:29.090624429Z"
+        }
+      }
+    }
+  ]
+}
+----
+
+NOTE: Create the index pipeline using the `PUT` command in the previous section before using the simulate pipeline API.
+
+[discrete]
+[[logs-stream-index-template]]
+==== Configure your data stream with an index template
+
+After creating your ingest pipeline, create an index template to point your log data to your pipeline using this command:
+
+[source,console]
+----
+PUT _index_template/logs-example-default-template
```

Comment:
@felixbarny @felixbarny If someone is building their own dataset, I assume they should not use `@custom` but the (future) `@default`?

Reply:
IIUC, this example is about registering a custom index templates with higher priority to override the default one. In such cases, neither `@custom` nor `@default` should be used and the only recommendation is to avoid using `@`. If that's the case, this looks right as it is

Reply:
What the user is building here is their custom dataset without having an instance of an integration. My assumption, a user could convert it to it eventually. 

As it is the user owned dataset, I'm tempted to recommend to use `@default`. It means, if someone else would want to extend the dataset could use `@custom`. 

> In such cases, neither @custom nor @default should be used and the only recommendation is to avoid using @. If that's the case, this looks right as it is

@eyalkoren In the context of the data stream naming scheme, should "random" templates ever be used? @joshdover How would Fleet keep track of these?

@weltenwort Also bringing you in here as I think this discussion could become interesting in the context of creating integrations.

@mdbirnstiehl Please don't block on this conversation, we can always update it.

Reply:
> @eyalkoren In the context of the data stream naming scheme, should "random" templates ever be used?

Again, my understanding is that this depends on who owns/registers the templates. The way I see it, the naming scheme is only relevant for templates managed by Elastic (never mind which component). This frees us from keeping track of anything else. We make best efforts to prevent clashes **between custom components and such managed by Elastic** by documenting that `@` should be avoided, but we are not eligible for preventing clashes between custom components. I hope this makes sense.

Reply:
In the future, I think that ideally users shouldn't need to create an index template but rather hook into extension points for the managed index templates. See also https://github.com/elastic/elasticsearch/issues/97664.

But for the time being, creating a custom `logs-example-default-template` is fine. The only thing I'd change here is to use the built-in index template for `logs-*-*` as a foundation for the custom index template. The built-in template can be retrieved via `GET _index_template/logs`.

Reply:
> How would Fleet keep track of these?

We haven't designed how someone would do this conversion from a existing index template and ingest pipeline to a proper package, so I can only speculate at this point.

The way I'd imagine it working though is that we could extract any settings defined at the top level and move them into the `@package` settings later (strictly speaking, we'd put them the package-spec definition, in the data stream's `manifest.yml` `elasticsearch.settings` block)

Reply:
I think there should be an option (eventually) to add a dataset to an integration. Not sure we can make this work if the dataset itself doesn't already follow the convention around naming.

@weltenwort Have you thought more about this?

Reply:
I tend to agree with what @joshdover hinted at:

- Guidance for the "manual quick-start" approach avoids any names that might collide with integration-managed assets (i.e. avoid `@`s).
- The "uplift into an integration" process would pick settings from the user-created assets and use them to create proper integration-managed assets. If those are ambiguous or conflicting we could let the user choose during the workflow.

The advantage of this approach would be that we don't front-load all the naming knowledge, which could increase the hurdle to get data in. It would also mean the instructions wouldn't have to change frequently as our conventions evolve. We "just" need to adapt our conversion routine.